[Lens] move `reference` handling to server

nickofthyme opened the pull request:
## Summary

Move Lens `reference` handling to server side.

Close https://github.com/elastic/kibana/issues/221943

## Details

Lens no longer relies on `references` from the `buildEmbeddable`. We now inject/extract references on the server.

Lens duplicates `references` in it's own serialized state so we just move them there and add the `savedObjectId` for by-ref Lens panels. Lens still inject these `references` internally by type across the Lens code.

For now we still store the `references` duplicated in the Lens SO state, but we can clean this up when we extract all the internal reference handling.

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

nickofthyme committed:
[Lens] move references to server

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
chore: cleanup transforms, handle dynamic actions, stub out api config transforms

nickofthyme committed:
Merge branch 'main' into lens-server-refs

elasticmachine commented:
Pinging @elastic/kibana-visualizations (Team:Visualizations)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/353341)
* Commit: 61822ed88674771db0df02c64aff2918e8b2daaf

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/353341#019a08d3-3ada-4552-8043-d9457a39f8a5)

### History
* :broken_heart: [Build #351156](https://buildkite.com/elastic/kibana-pull-request/builds/351156) failed 704035413d7aea64570af5c92729f68668fea8a8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353341","number":353341,"commit":"61822ed88674771db0df02c64aff2918e8b2daaf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351156","number":351156,"commit":"704035413d7aea64570af5c92729f68668fea8a8"}],"number":353341}
buildkite-pr-comment-->

nickofthyme committed:
fix type errors

nickofthyme committed:
Merge branch 'main' into lens-server-refs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/353807)
* Commit: b72c7e6f08941fe95a9ab5d81d5952b607412da3

### Failed CI Steps
* [Jest Tests #14](https://buildkite.com/elastic/kibana-pull-request/builds/353807#019a0db3-4290-4a14-bb42-0b144fcd02b8)
* [FTR Configs #70](https://buildkite.com/elastic/kibana-pull-request/builds/353807#019a0db3-42dd-40ae-882b-84ca27189329)
* [FTR Configs #76](https://buildkite.com/elastic/kibana-pull-request/builds/353807#019a0db3-42e3-46e5-93da-fd3eb21df7da)
* [FTR Configs #141](https://buildkite.com/elastic/kibana-pull-request/builds/353807#019a0db3-4327-48f1-adb8-c8c30f7aae69)
* [FTR Configs #141](https://buildkite.com/elastic/kibana-pull-request/builds/353807#019a0dc1-a763-443e-8534-4c2864882720)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353807#019a0db3-4290-4a14-bb42-0b144fcd02b8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353807/jobs/019a0db3-4290-4a14-bb42-0b144fcd02b8/artifacts/019a0dcb-8c98-48fd-bacd-15e6cce57ab1) Jest Tests #<span></span>14 / Dashboard services API serializeState should serialize state for a by-reference panel
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353807#019a0db3-4290-4a14-bb42-0b144fcd02b8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353807/jobs/019a0db3-4290-4a14-bb42-0b144fcd02b8/artifacts/019a0dcb-8c99-49bd-a713-5cdae47fc4fe) Jest Tests #<span></span>14 / Dashboard services API serializeState should work for a by-value panel
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353807#019a0db3-42e3-46e5-93da-fd3eb21df7da) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353807/jobs/019a0db3-42e3-46e5-93da-fd3eb21df7da/artifacts/019a0dc9-96fc-419a-9790-20da417f2689) FTR Configs #<span></span>76 / lens app - group 3 lens add-to-dashboards tests should allow new lens be added by value to an existing dashboard
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353807#019a0db3-4327-48f1-adb8-c8c30f7aae69) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353807/jobs/019a0db3-4327-48f1-adb8-c8c30f7aae69/artifacts/019a0dc1-759b-4711-9de1-88b1e0d2450e) FTR Configs #<span></span>141 / lens app - group 4 lens dashboard tests metric should be embeddable
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353807#019a0dc1-a763-443e-8534-4c2864882720) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353807/jobs/019a0dc1-a763-443e-8534-4c2864882720/artifacts/019a0dcb-d4b0-4a89-b094-ce2039a71ca7) FTR Configs #<span></span>141 / lens app - group 4 lens dashboard tests metric should be embeddable
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353807#019a0db3-42dd-40ae-882b-84ca27189329) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353807/jobs/019a0db3-42dd-40ae-882b-84ca27189329/artifacts/019a0dcf-8a46-419e-aa73-771e8b341564) FTR Configs #<span></span>70 / Search session sharing Search session sharing with lens should share search session with by value lens and don't share with by reference

### History
* :broken_heart: [Build #353341](https://buildkite.com/elastic/kibana-pull-request/builds/353341) failed 61822ed88674771db0df02c64aff2918e8b2daaf
* :broken_heart: [Build #351156](https://buildkite.com/elastic/kibana-pull-request/builds/351156) failed 704035413d7aea64570af5c92729f68668fea8a8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353807","number":353807,"commit":"b72c7e6f08941fe95a9ab5d81d5952b607412da3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353341","number":353341,"commit":"61822ed88674771db0df02c64aff2918e8b2daaf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351156","number":351156,"commit":"704035413d7aea64570af5c92729f68668fea8a8"}],"number":353807}
buildkite-pr-comment-->

dej611 commented on x-pack/platform/plugins/shared/lens/server/api/routes/utils.ts:
```diff
@@ -5,41 +5,80 @@
+    // TODO: fix these type issues
+    visualizationType,
+    title: attributes.title ?? '',
+    description: attributes.description ?? undefined,
```
```suggestion
    description: attributes.description,
```

dej611 commented on x-pack/platform/plugins/shared/lens/server/content_management/v1/schema/common.ts:
```diff
@@ -23,90 +23,37 @@ export const lensItemAttributesSchema = schema.object(
-  },
-  { unknowns: 'forbid' }
-);
+// TODO: cleanup data for update, should we forbid or just ignore body.id on update?
```
Let's discuss this offline

dej611 committed:
Merge branch 'main' into lens-server-refs

dej611 reviewed (approved):
Tested locally and works fine.
Left some minor comments, in particular about type locations.
Will approve to unblock.

dej611 committed:
Merge remote-tracking branch 'upstream/main' into lens-server-refs

dej611 committed:
:label: first type pass

dej611 committed:
:label: More types fixed

dej611 committed:
:wrench: Flag package for treeshake

dej611 committed:
:white_check_mark: Align test to lack of references

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/354562)
* Commit: 7319d75abbe73ea16b60478cdf2500ecd7511c31
* [Webpack Bundle Analyzer](https://ci-artifacts.kibana.dev/webpack_bundle_analyzer/pr-239029/7319d75abbe73ea16b60478cdf2500ecd7511c31)

### Failed CI Steps
* [FTR Configs #3](https://buildkite.com/elastic/kibana-pull-request/builds/354562#019a1689-e882-49ab-a56b-57ba6d52f515)
* [FTR Configs #82](https://buildkite.com/elastic/kibana-pull-request/builds/354562#019a1689-e8d0-431d-a8ef-faabf38ed021)
* [FTR Configs #82](https://buildkite.com/elastic/kibana-pull-request/builds/354562#019a169b-f573-4853-a541-6f77116bccc1)
* [FTR Configs #99](https://buildkite.com/elastic/kibana-pull-request/builds/354562#019a1689-e8e0-44ee-b49c-f1a374615ff2)
* [FTR Configs #99](https://buildkite.com/elastic/kibana-pull-request/builds/354562#019a1697-cac8-476f-8ad6-1cb9d64509ae)
* [FTR Configs #130](https://buildkite.com/elastic/kibana-pull-request/builds/354562#019a1689-e901-4bb4-b71f-d8ad23591d7d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354562#019a1689-e901-4bb4-b71f-d8ad23591d7d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354562/jobs/019a1689-e901-4bb4-b71f-d8ad23591d7d/artifacts/019a16a5-1e79-4a84-a68d-9c7fdde878c9) FTR Configs #<span></span>130 / Dashboard Session and searches integration Slow lens with other bucket Other bucket should be added to a session when restoring
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354562#019a1689-e8d0-431d-a8ef-faabf38ed021) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354562/jobs/019a1689-e8d0-431d-a8ef-faabf38ed021/artifacts/019a169b-e412-4935-8d91-8056589eeac9) FTR Configs #<span></span>82 / lens app - group 3 lens add-to-dashboards tests should allow new lens be added by value to an existing dashboard
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354562#019a169b-f573-4853-a541-6f77116bccc1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354562/jobs/019a169b-f573-4853-a541-6f77116bccc1/artifacts/019a16a5-68f7-4f98-9b76-82660bbddcbb) FTR Configs #<span></span>82 / lens app - group 3 lens add-to-dashboards tests should allow new lens be added by value to an existing dashboard
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354562#019a1689-e8e0-44ee-b49c-f1a374615ff2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354562/jobs/019a1689-e8e0-44ee-b49c-f1a374615ff2/artifacts/019a1697-b731-45ba-aeaf-873d96ce788a) FTR Configs #<span></span>99 / lens app - group 4 lens dashboard tests metric should be embeddable
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354562#019a1697-cac8-476f-8ad6-1cb9d64509ae) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354562/jobs/019a1697-cac8-476f-8ad6-1cb9d64509ae/artifacts/019a16a2-dfb5-4ceb-9ba1-2934dfa01c2f) FTR Configs #<span></span>99 / lens app - group 4 lens dashboard tests metric should be embeddable

### History
* :broken_heart: [Build #353909](https://buildkite.com/elastic/kibana-pull-request/builds/353909) failed 642db890d8095f2dd5cf22ac4310dedfd8e32422
* :broken_heart: [Build #353807](https://buildkite.com/elastic/kibana-pull-request/builds/353807) failed b72c7e6f08941fe95a9ab5d81d5952b607412da3
* :broken_heart: [Build #353341](https://buildkite.com/elastic/kibana-pull-request/builds/353341) failed 61822ed88674771db0df02c64aff2918e8b2daaf
* :broken_heart: [Build #351156](https://buildkite.com/elastic/kibana-pull-request/builds/351156) failed 704035413d7aea64570af5c92729f68668fea8a8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354562","number":354562,"commit":"7319d75abbe73ea16b60478cdf2500ecd7511c31"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353909","number":353909,"commit":"642db890d8095f2dd5cf22ac4310dedfd8e32422"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353807","number":353807,"commit":"b72c7e6f08941fe95a9ab5d81d5952b607412da3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353341","number":353341,"commit":"61822ed88674771db0df02c64aff2918e8b2daaf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351156","number":351156,"commit":"704035413d7aea64570af5c92729f68668fea8a8"}],"number":354562}
buildkite-pr-comment-->

nickofthyme committed:
chore: remove `schema` on transform definition

nickofthyme committed:
chore: move `LensParentApi` types to `@kbn/lens-common`

nickofthyme committed:
chore: revert code cleanup

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
fix: lens api integration tests

nickofthyme commented on x-pack/platform/plugins/shared/lens/common/transforms/index.ts:
```diff
@@ -5,4 +5,23 @@
+    transformIn: getTransformIn(deps),
+    transformOut: getTransformOut(deps),
+    transformOutInjectsReferences: true,
+    schema: schema.any(),
```
Ya good idea, removed in 19a76e7351522f8eafcf46b61b8073f0b8414d8c

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

nickofthyme committed:
fix add panel from library

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
Merge remote-tracking branch 'origin/lens-server-refs' into lens-server-refs

nickofthyme commented on src/platform/packages/shared/kbn-lens-common/embeddable/types.ts:
The last remaining failing functional test is a pain. The background search session makes Lens not render in one case. The test runs fine in `main` but has been flaky https://github.com/elastic/kibana/issues/239154.

nickofthyme committed:
fix: background session issue

bg sessions use the public url transforms which defaults to references from the dashboard url which are now always empty, soon to be removed entirely.

But in Lens we had logic to filter references from the dashboard based on references from the lens SO. Instead we should just merge all refs from the lens SO and dashboard.

nickofthyme committed:
Merge branch 'main' into lens-server-refs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/356443)
* Commit: 7b2483a6378acb6e334716f15a32f717d89c1cac
* [Webpack Bundle Analyzer](https://ci-artifacts.kibana.dev/webpack_bundle_analyzer/pr-239029/7b2483a6378acb6e334716f15a32f717d89c1cac)

### Failed CI Steps
* [FTR Configs #109](https://buildkite.com/elastic/kibana-pull-request/builds/356443#019a31af-d2ac-4a83-bd52-4a77785520cc)
* [FTR Configs #109](https://buildkite.com/elastic/kibana-pull-request/builds/356443#019a31df-2bef-4160-9910-76627dbce733)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/356443#019a31af-d2ac-4a83-bd52-4a77785520cc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/356443/jobs/019a31af-d2ac-4a83-bd52-4a77785520cc/artifacts/019a31df-0a34-4100-b6bb-1c58dcceaf27) FTR Configs #<span></span>109 / serverless search UI - search features Search Playground - Saved Playgrounds Update Saved Playground should allow updating playground name
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/356443#019a31df-2bef-4160-9910-76627dbce733) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/356443/jobs/019a31df-2bef-4160-9910-76627dbce733/artifacts/019a31f3-313b-4432-ac84-46aca0036579) FTR Configs #<span></span>109 / serverless search UI - search features Search Playground - Saved Playgrounds Update Saved Playground should allow updating playground name

### History
* :broken_heart: [Build #356004](https://buildkite.com/elastic/kibana-pull-request/builds/356004) failed 7ee482fc95aac5c2afc89e4c5ccbd4654e4646d6
* :broken_heart: [Build #355553](https://buildkite.com/elastic/kibana-pull-request/builds/355553) failed 8b2a35e3e1a15a86e4968a2c62e8816791bf7204
* :broken_heart: [Build #354562](https://buildkite.com/elastic/kibana-pull-request/builds/354562) failed 7319d75abbe73ea16b60478cdf2500ecd7511c31
* :broken_heart: [Build #353909](https://buildkite.com/elastic/kibana-pull-request/builds/353909) failed 642db890d8095f2dd5cf22ac4310dedfd8e32422
* :broken_heart: [Build #353807](https://buildkite.com/elastic/kibana-pull-request/builds/353807) failed b72c7e6f08941fe95a9ab5d81d5952b607412da3

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356443","number":356443,"commit":"7b2483a6378acb6e334716f15a32f717d89c1cac"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356004","number":356004,"commit":"7ee482fc95aac5c2afc89e4c5ccbd4654e4646d6"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355553","number":355553,"commit":"8b2a35e3e1a15a86e4968a2c62e8816791bf7204"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354562","number":354562,"commit":"7319d75abbe73ea16b60478cdf2500ecd7511c31"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353909","number":353909,"commit":"642db890d8095f2dd5cf22ac4310dedfd8e32422"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353807","number":353807,"commit":"b72c7e6f08941fe95a9ab5d81d5952b607412da3"}],"number":356443}
buildkite-pr-comment-->

nickofthyme committed:
Merge branch 'main' into lens-server-refs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/356479)
* Commit: 897b867ab7b1eb94e29ffb4528b41ec375b0e83a

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/356479#019a3200-cdfd-4e95-aa55-f3994b54ed72)

### History
* :broken_heart: [Build #356443](https://buildkite.com/elastic/kibana-pull-request/builds/356443) failed 7b2483a6378acb6e334716f15a32f717d89c1cac
* :broken_heart: [Build #356004](https://buildkite.com/elastic/kibana-pull-request/builds/356004) failed 7ee482fc95aac5c2afc89e4c5ccbd4654e4646d6
* :broken_heart: [Build #355553](https://buildkite.com/elastic/kibana-pull-request/builds/355553) failed 8b2a35e3e1a15a86e4968a2c62e8816791bf7204
* :broken_heart: [Build #354562](https://buildkite.com/elastic/kibana-pull-request/builds/354562) failed 7319d75abbe73ea16b60478cdf2500ecd7511c31
* :broken_heart: [Build #353909](https://buildkite.com/elastic/kibana-pull-request/builds/353909) failed 642db890d8095f2dd5cf22ac4310dedfd8e32422

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356479","number":356479,"commit":"897b867ab7b1eb94e29ffb4528b41ec375b0e83a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356443","number":356443,"commit":"7b2483a6378acb6e334716f15a32f717d89c1cac"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356004","number":356004,"commit":"7ee482fc95aac5c2afc89e4c5ccbd4654e4646d6"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355553","number":355553,"commit":"8b2a35e3e1a15a86e4968a2c62e8816791bf7204"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354562","number":354562,"commit":"7319d75abbe73ea16b60478cdf2500ecd7511c31"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353909","number":353909,"commit":"642db890d8095f2dd5cf22ac4310dedfd8e32422"}],"number":356479}
buildkite-pr-comment-->

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
chore: update cloud limits

nickofthyme committed:
Merge remote-tracking branch 'origin/lens-server-refs' into lens-server-refs

nickofthyme committed:
Merge branch 'main' into lens-server-refs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/356508)
* Commit: c364724270ce433f6ec7207ff3fb2f7001708729

### Failed CI Steps
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/356508#019a32c2-3523-447b-af16-8e7d5db6ed1d)

### History
* :broken_heart: [Build #356501](https://buildkite.com/elastic/kibana-pull-request/builds/356501) failed 35c3697237ea08a996a83bb4c410fab29b01b979
* :broken_heart: [Build #356479](https://buildkite.com/elastic/kibana-pull-request/builds/356479) failed 897b867ab7b1eb94e29ffb4528b41ec375b0e83a
* :broken_heart: [Build #356443](https://buildkite.com/elastic/kibana-pull-request/builds/356443) failed 7b2483a6378acb6e334716f15a32f717d89c1cac
* :broken_heart: [Build #356004](https://buildkite.com/elastic/kibana-pull-request/builds/356004) failed 7ee482fc95aac5c2afc89e4c5ccbd4654e4646d6
* :broken_heart: [Build #355553](https://buildkite.com/elastic/kibana-pull-request/builds/355553) failed 8b2a35e3e1a15a86e4968a2c62e8816791bf7204

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356508","number":356508,"commit":"c364724270ce433f6ec7207ff3fb2f7001708729"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356501","number":356501,"commit":"35c3697237ea08a996a83bb4c410fab29b01b979"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356479","number":356479,"commit":"897b867ab7b1eb94e29ffb4528b41ec375b0e83a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356443","number":356443,"commit":"7b2483a6378acb6e334716f15a32f717d89c1cac"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356004","number":356004,"commit":"7ee482fc95aac5c2afc89e4c5ccbd4654e4646d6"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355553","number":355553,"commit":"8b2a35e3e1a15a86e4968a2c62e8816791bf7204"}],"number":356508}
buildkite-pr-comment-->

dej611 committed:
Merge branch 'main' into lens-server-refs
